### PR TITLE
homeassistant-satellite: init at 2.3.0

### PR DESCRIPTION
**Pull request by mweinelt**
## Description of changes

Packages up a few things

- pasimple
  - Play/record audio via PulseAudio/PipeWire. A PulseAudio "simple API" wrapper.
- homeassistant-satellite
  - Streaming audio satellite for Home Assistant voice pipelines.

Creates a module for the homeassistant-satellite.
- [x] Tested with `pulseaudio.enable` against `pipewire-pulse`.

Relaxes hardening on the wyoming-openwakeword module.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/home-automation/homeassistant-satellite.nix [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+  # override the package with the relevant vad dependencies
+  package = cfg.package.overridePythonAttrs (oldAttrs: {
+    propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++
+      (if cfg.vad == "webrtcvad" then
+        cfg.package.optional-dependencies.webrtc
+      else if cfg.vad == "silero" then
+        cfg.package.optional-dependencies.silerovad
+      else []);
+  });
```
Comment (mweinelt) [inline nixos/modules/services/home-automation/homeassistant-satellite.nix:32]: Needs to take pulseaudio dependencies into account.

File: nixos/modules/services/home-automation/homeassistant-satellite.nix [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      ];
+      serviceConfig = {
+        DynamicUser = true;
+        User = "homeassistant-satellite";
+        # https://github.com/rhasspy/hassio-addons/blob/master/assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/run
+        ExecStart = ''
+          ${package}/bin/homeassistant-satellite \
+            --host ${cfg.host} \
+            --port ${cfg.port} \
+            --token "$(cat ${cfg.token})" \
```
Comment (mweinelt) [inline nixos/modules/services/home-automation/homeassistant-satellite.nix:112]:
```suggestion
            --token-file ${cfg.tokenFile} \
```

File: nixos/modules/services/home-automation/homeassistant-satellite.nix [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      '';
+      apply = toString;
+    };
+
+    tokenFile = mkOption {
+      type = path;
+      example = "/run/keys/hass-token";
+      description = mdDoc ''
+        Path to a file containing a Long-lived access token for your Home Assistant instance.
+      '';
```
Comment (mweinelt) [inline nixos/modules/services/home-automation/homeassistant-satellite.nix:66]:
```suggestion
      '';
      apply = escapeShellArgs;
```

File: nixos/modules/services/home-automation/homeassistant-satellite.nix [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        Port on which your Home Assistance can be reached.
+      '';
+      apply = toString;
+    };
+
+    tokenFile = mkOption {
+      type = path;
+      example = "/run/keys/hass-token";
+      description = mdDoc ''
+        Path to a file containing a Long-lived access token for your Home Assistant instance.
```
Comment (fabaff) [inline nixos/modules/services/home-automation/homeassistant-satellite.nix:65]:
```suggestion
        Path to a file containing a long-lived access token for your Home Assistant instance.
```

File: pkgs/by-name/ho/homeassistant-satellite/package.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  pythonImportsCheck = [ "homeassistant_satellite" ];
+
+  # no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Streaming audio satellite for Home Assistant";
+    homepage = "https://github.com/synesthesiam/homeassistant-satellite";
```
Comment (fabaff) [inline pkgs/by-name/ho/homeassistant-satellite/package.nix:48]:
```suggestion
    homepage = "https://github.com/synesthesiam/homeassistant-satellite";
    changelog = "https://github.com/synesthesiam/homeassistant-satellite/blob/v${version}/CHANGELOG.md";
```

**Pull request merged**